pkg/asset/imagebased/image/imagebased_config.go

```diff
@@ -0,0 +1,409 @@
+		}
+
+		for midx, mirror := range group.Mirrors {
+			if err := validateNamedRepository(mirror); err != nil {
+				allErrs = append(allErrs, field.Invalid(groupf.Child("mirrors").Index(midx), mirror, err.Error()))
+				continue
+			}
+		}
+	}
+	return allErrs
+}
+
+func (i *ImageBasedInstallationConfig) validateImageDigestSources() field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	fldPath := field.NewPath("imageDigestSources")
+
+	for gidx, group := range i.Config.ImageDigestSources {
+		groupf := fldPath.Index(gidx)
+		if err := validateNamedRepository(group.Source); err != nil {
+			allErrs = append(allErrs, field.Invalid(groupf.Child("source"), group.Source, err.Error()))
+		}
+
+		for midx, mirror := range group.Mirrors {
+			if err := validateNamedRepository(mirror); err != nil {
+				allErrs = append(allErrs, field.Invalid(groupf.Child("mirrors").Index(midx), mirror, err.Error()))
+				continue
+			}
+		}
+	}
+	return allErrs
+}
+
+func validateNamedRepository(r string) error {
+	ref, err := dockerref.ParseNamed(r)
+	if err != nil {
+		// If a mirror name is provided without the named reference,
+		// then the name is not considered canonical and will cause
+		// an error. e.g. registry.lab.redhat.com:5000 will result
+		// in an error. Instead we will check whether the input is
+		// a valid hostname as a workaround.
+		if errors.Is(err, dockerref.ErrNameNotCanonical) {
+			// If the hostname string contains a port, lets attempt
+			// to split them
+			host, _, err := net.SplitHostPort(r)
+			if err != nil {
+				host = r
+			}
+			if err = validate.Host(host); err != nil {
+				return fmt.Errorf("the repository provided is invalid: %w", err)
+			}
+			return nil
+		}
+		return fmt.Errorf("failed to parse: %w", err)
+	}
+	if !dockerref.IsNameOnly(ref) {
+		return errors.New("must be repository--not reference")
+	}
+	return nil
+}
+
+func (i *ImageBasedInstallationConfig) validateProxy() field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	fldPath := field.NewPath("proxy")
+
+	// empty Proxy is fine
+	if i.Config.Proxy == nil {
+		return nil
+	}
+
+	if i.Config.Proxy.HTTPProxy == "" && i.Config.Proxy.HTTPSProxy == "" {
+		allErrs = append(allErrs, field.Required(fldPath, "must include httpProxy or httpsProxy"))
+	}
+
+	if i.Config.Proxy.HTTPProxy != "" {
+		allErrs = append(allErrs, validateURI(i.Config.Proxy.HTTPProxy, fldPath.Child("httpProxy"), []string{"http"})...)
+	}
+	if i.Config.Proxy.HTTPSProxy != "" {
+		allErrs = append(allErrs, validateURI(i.Config.Proxy.HTTPSProxy, fldPath.Child("httpsProxy"), []string{"http", "https"})...)
+	}
+	if i.Config.Proxy.NoProxy != "" && i.Config.Proxy.NoProxy != "*" {
+		if strings.Contains(i.Config.Proxy.NoProxy, " ") {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("noProxy"), i.Config.Proxy.NoProxy, "noProxy must not have spaces"))
+		}
+		for idx, v := range strings.Split(i.Config.Proxy.NoProxy, ",") {
+			v = strings.TrimSpace(v)
```

Comment:
in line 380 we are validating that there is no space in NoProxy but here we trimming spaces, maybe we can trim all spaces and remove "no space" validation ?